boards/raspberry_pi_pico_w/src/main.rs

```diff
@@ -0,0 +1,213 @@
+>;
+
+type WifiDriver = capsules_extra::wifi::WifiDriver<'static, CYW4343xHw>;
+
+/// Supported drivers by the platform
+pub struct RaspberryPiPicoW {
+    base: raspberry_pi_pico::Platform,
+    wifi: &'static WifiDriver,
+}
+
+impl SyscallDriverLookup for RaspberryPiPicoW {
+    fn with_driver<F, R>(&self, driver_num: usize, f: F) -> R
+    where
+        F: FnOnce(Option<&dyn SyscallDriver>) -> R,
+    {
+        match driver_num {
+            capsules_extra::wifi::DRIVER_NUM => f(Some(self.wifi)),
+            _ => self.base.with_driver(driver_num, f),
+        }
+    }
+}
+
+impl KernelResources<Rp2040<'static, Rp2040DefaultPeripherals<'static>>> for RaspberryPiPicoW {
+    type SyscallDriverLookup = Self;
+    type SyscallFilter = ();
+    type ProcessFault = ();
+    type Scheduler = RoundRobinSched<'static>;
+    type SchedulerTimer = cortexm0p::systick::SysTick;
+    type WatchDog = ();
+    type ContextSwitchCallback = ();
+
+    fn syscall_driver_lookup(&self) -> &Self::SyscallDriverLookup {
+        self
+    }
+    fn syscall_filter(&self) -> &Self::SyscallFilter {
+        &()
+    }
+    fn process_fault(&self) -> &Self::ProcessFault {
+        &()
+    }
+    fn scheduler(&self) -> &Self::Scheduler {
+        self.base.scheduler
+    }
+    fn scheduler_timer(&self) -> &Self::SchedulerTimer {
+        &self.base.systick
+    }
+    fn watchdog(&self) -> &Self::WatchDog {
+        &()
+    }
+    fn context_switch_callback(&self) -> &Self::ContextSwitchCallback {
+        &()
+    }
+}
+
+/// This is in a separate, inline(never) function so that its stack frame is
+/// removed when this function returns. Otherwise, the stack space used for
+/// these static_inits is wasted.
+#[inline(never)]
+pub unsafe fn start() -> (
+    &'static kernel::Kernel,
+    RaspberryPiPicoW,
+    &'static rp2040::chip::Rp2040<'static, Rp2040DefaultPeripherals<'static>>,
+) {
+    let output = raspberry_pi_pico::Output::Cdc;
+
+    // Uncomment this to use UART0 as output for console caspule/debug writer/process console
+    // let output = raspberry_pi_pico::Output::Uart;
+
+    let (board_kernel, base, peripherals, mux_alarm, chip) = raspberry_pi_pico::setup(output);
+
+    // Set the UART used for panic
+    (*core::ptr::addr_of_mut!(io::WRITER)).set_uart(&peripherals.uart0);
+
+    // WIFI
+
+    let cs = peripherals.pins.get_pin(RPGpio::GPIO25);
+    cs.make_output();
+
+    let pio_gspi = PioGspiComponent::new(
+        &peripherals.pio0,
+        pio::SMNumber::SM0,
+        peripherals.dma.channel(dma::Channel::Channel0),
+        dma::Irq::Irq0,
+        RPGpio::GPIO29,
+        RPGpio::GPIO24,
+        cs,
+    )
+    .finalize(pio_gpsi_component_static!());
+
+    let (fw, nvram, clm) = (
+        tock_firmware_cyw43::cyw43439::FW,
+        tock_firmware_cyw43::cyw43439::NVRAM,
+        tock_firmware_cyw43::cyw43439::CLM,
+    );
+
+    let pwr = peripherals.pins.get_pin(RPGpio::GPIO23);
+    pwr.make_output();
+
+    let bus =
```

Comment:
Nit, but can we rename this to something more meaningful than `bus`? Perhaps `cyw4343_spi_bus`?